### PR TITLE
Set CrossGenTest to false for tests blocking green CI

### PR DESCRIPTION
**Pull request by echesakovMSFT**
For tests failing due to https://github.com/dotnet/coreclr/issues/23650 and https://github.com/dotnet/coreclr/issues/23643

### CODE REVIEW

**Comment by echesakovMSFT** [comment]
@swaroop-sridhar @fadimounir PTAL

**Comment by echesakovMSFT** [comment]
cc @dotnet/jit-contrib

**Comment by fadimounir** [comment]
it looks good, but out of curiosity, why do we run mainv1 and mainv2 in the R2R test mode? Aren't these already crossgen R2R tests?

**Comment by echesakovMSFT** [comment]
@fadimounir Okay - now I see that we have custom logic in mainv1.csproj and mainv2.csproj files, so my changes in this PR wouldn't help much

```xml
  <PropertyGroup>
    <CLRTestBatchPreCommands><![CDATA[
$(CLRTestBatchPreCommands)
DEL test.dll
COPY /Y ..\testv1\test\test.dll test.dll
%Core_Root%\crossgen /readytorun /platform_assemblies_paths %Core_Root%%3B%25CD% /out test.ni.dll test.dll
%Core_Root%\crossgen /readytorun /platform_assemblies_paths %Core_Root%%3B%25CD% /out fieldgetter.ni.dll fieldgetter.dll
%Core_Root%\crossgen /readytorun /platform_assemblies_paths %Core_Root%%3B%25CD% /out mainv1.ni.exe mainv1.exe
]]></CLRTestBatchPreCommands>
  <BashCLRTestPreCommands><![CDATA[
$(BashCLRTestPreCommands)
rm -f test.dll
cp ../testv1/test/test.dll test.dll
$CORE_ROOT/crossgen -readytorun -platform_assemblies_paths $CORE_ROOT:`pwd` -out test.ni.dll test.dll
$CORE_ROOT/crossgen -readytorun -platform_assemblies_paths $CORE_ROOT:`pwd` -out fieldgetter.ni.dll fieldgetter.dll
$CORE_ROOT/crossgen -readytorun -platform_assemblies_paths $CORE_ROOT:`pwd` -out mainv1.ni.exe mainv1.exe
]]></BashCLRTestPreCommands>
```

To answer your question

> it looks good, but out of curiosity, why do we run mainv1 and mainv2 in the R2R test mode? Aren't these already crossgen R2R tests?

We usually run all the test in both modes (R2R and non-R2R)

**Comment by fadimounir** [comment]
I feel like these 2 tests should be excluded from the R2R test mode, since they are special in that they run crossgen explicitly as part of normal test run.

**Comment by echesakovMSFT** [comment]
@fadimounir Interesting that these tests fail when framework is crossgen-d and don't fail when framework is IL

File: tests/src/readytorun/tests/mainv1.csproj [hunk 1]
```diff
@@ -11,6 +11,8 @@
     <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\..\</SolutionDir>
     <CLRTestKind>BuildAndRun</CLRTestKind>
     <DefineConstants>$(DefineConstants);STATIC;CORECLR</DefineConstants>
+    <!-- https://github.com/dotnet/coreclr/issues/23643 -->
+    <CrossGenTest Condition=" '$(BuildOS)' == 'Windows_NT' And ('$(BuildArch)' == 'x64' Or '$(BuildArch)' == 'x86') ">false</CrossGenTest>
```
Comment (fadimounir) [inline tests/src/readytorun/tests/mainv1.csproj]: Refering to my comment, i think this should be set to false unconditionally to exclude the test from the R2R mode, given how special this test is

**Comment by fadimounir** [comment]
@echesakovMSFT I actually meant remove the condition for all tests :). I'm working on some changes to enable my pinvokes on Unix as well

**Comment by echesakovMSFT** [comment]
> @echesakovMSFT I actually meant remove the condition for all tests :). I'm working on some changes to enable my pinvokes on Unix as well

I can do this but I am wondering why we are removing the conditions for Interop tests?

**Comment by fadimounir** [comment]
> I can do this but I am wondering why we are removing the conditions for Interop tests?

The pinvokes causing the failure are not architecture specific, so we don't need a condition on the architecture. Also, they are just enabled on Windows now, but i'm planning on enabling that for Unix, so instead of adding a Windows_NT condition and having the test fail again and have to refix this, it's better to just disable them unconditionally for now

**Comment by fadimounir** [comment]
@echesakovMSFT Can you please hold of from merging this? I'm working on #23828 and it will fix these failures

**Comment by echesakovMSFT** [comment]
@fadimounir Sure

File: tests/src/Interop/NativeLibraryResolveCallback/CallbackTests.csproj [hunk 2]
```diff
@@ -12,6 +12,8 @@
     <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\..\</SolutionDir>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <ReferenceSystemPrivateCoreLib>true</ReferenceSystemPrivateCoreLib>
+    <!-- https://github.com/dotnet/coreclr/issues/23650 -->
+    <CrossGenTest>false</CrossGenTest>
```
Comment (fadimounir) [inline tests/src/Interop/NativeLibraryResolveCallback/CallbackTests.csproj]: Undo the changes in this file. Fixed by #23828

File: tests/src/Interop/NativeLibraryResolveEvent/ResolveEventTests.csproj [hunk 3]
```diff
@@ -14,6 +14,8 @@
     <ReferenceSystemPrivateCoreLib>true</ReferenceSystemPrivateCoreLib>
     <!-- The test leaves AssemblyLoadContext.Default.ResolvingUnmanagedDll event registered to a method in the test at exit -->
     <UnloadabilityIncompatible>true</UnloadabilityIncompatible>
+    <!-- See https://github.com/dotnet/coreclr/issues/23650 -->
+    <CrossGenTest>false</CrossGenTest>
```
Comment (fadimounir) [inline tests/src/Interop/NativeLibraryResolveEvent/ResolveEventTests.csproj]: Undo the changes in this file. Fixed by #23828

**Comment by fadimounir** [comment]
Thanks for making this change @echesakovMSFT
